braindecode/preprocessing/preprocess.py

```diff
@@ -120,54 +127,123 @@ def preprocess(concat_ds, preprocessors):
         A concat of BaseDataset or WindowsDataset datasets to be preprocessed.
     preprocessors: list(Preprocessor)
         List of Preprocessor objects to apply to the dataset.
+    save_dir : str | None
+        If a string, the preprocessed data will be saved under the specified
+        directory and the datasets in ``concat_ds`` will be reloaded with
+        `preload=False`.
+    overwrite : bool
+        When `save_dir` is provided, controls whether to overwrite existing
+        files in `save_dir`.
```

Comment:
Overwriting logic also changed. We will delete and create new subdirectories if overwrite = True

Reply:
Is the doc string here now consistent with what happens after merging #288 @hubertjb ?